gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -279,7 +280,8 @@ private void publishTaskData() throws IOException {
       @SuppressWarnings("unchecked")
       Class<? extends DataPublisher> dataPublisherClass = (Class<? extends DataPublisher>) Class.forName(
           this.taskState.getProp(ConfigurationKeys.DATA_PUBLISHER_TYPE, ConfigurationKeys.DEFAULT_DATA_PUBLISHER_TYPE));
-      DataPublisher publisher = closer.register(DataPublisher.getInstance(dataPublisherClass, this.taskState));
+      SingleTaskDataPublisher publisher =
```

Comment:
Ok, I see this is only used if `publish.data.at.job.level` is false. Could you add a more descriptive error message to the catch block? Maybe mentions that publishing data at task level is only supported for publishers extending `SingleTaskDataPublisher`?